document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -445,6 +429,58 @@ public boolean bulkUpsert(Map<Key, Document> documents) {
     return false;
   }
 
+  private int[] bulkUpsertImpl(Map<Key, Document> documents) throws SQLException, IOException {
+    PreparedStatement preparedStatement = client
+        .prepareStatement(getUpsertSQL(), Statement.RETURN_GENERATED_KEYS);
+    for (Map.Entry<Key, Document> entry : documents.entrySet()) {
+
+      Key key = entry.getKey();
+      String jsonString = prepareUpsertDocument(key, entry.getValue());
+
+      preparedStatement.setString(1, key.toString());
+      preparedStatement.setString(2, jsonString);
+      preparedStatement.setString(3, jsonString);
+
+      preparedStatement.addBatch();
+    }
+
+    return preparedStatement.executeBatch();
+  }
+
+  @Override
+  public Iterator<Document> bulkUpsertAndReturnOlderDocuments(Map<Key, Document> documents) throws IOException {
+    try {
+      String collect = documents.keySet().stream()
+          .map(val -> "'" + val.toString() + "'")
+          .collect(Collectors.joining(", "));
+
+      String space = " ";
+      String query = new StringBuilder("SELECT * FROM")
```

Comment:
I usually have reluctancies with the `select *`, do we really need to pass it all, can't we make explicit the fields we pass so that changes in the schema don't affect this kind of methods silently?

Reply:
@jcchavezs The need here is indeed the full document since we want to return all the fields of the document's current copy, as the API requires it.
At this layer, we don't know what all fields are present in the document actually.